packages/tx/src/eip2930Transaction.ts

```diff
@@ -188,6 +189,10 @@ export default class AccessListEIP2930Transaction extends BaseTransaction<Access
 
     this.gasPrice = new BN(toBuffer(gasPrice === '' ? '0x' : gasPrice))
 
+    if (this.gasPrice.mul(this.gasLimit).gt(MAX_INTEGER)) {
+      throw new Error('gas limit * price overflow')
+    }
```

Comment:
[This test covers this scenario in the ethereum-tests repo])https://github.com/ethereum/tests/blob/develop/TransactionTests/ttGasLimit/TransactionWithGasLimitxPriceOverflow.json).  Do you want us to have one in our own test suite as well?

Reply:
Ah ok, not sure, maybe that's ok then.

Reply:
i was also thinking about this yesterday, in general i think it is always nice to add coverage in our own test suite with any new lines added to source so we can have more local coverage, otherwise our overall repo coverage will continue to decline over time.

we could also consider adding the ethereum-tests transactionRunner to the coverage step if that doesn't add too much time and does also improve the coverage with these more isolated errors/validation.

Reply:
Okay, I've added some tests to cover the scenarios Holger outlined so will see where we land once we get final feedback on the new leadingZeroes function.

Reply:
i'm still seeing a check warning "Added line #L207 was not covered by tests"

Reply:
ah looks like the one below in baseTransaction is covered but this eip2930Transaction isn't, that's probably fine